main/autogen/autogen.cc

```diff
@@ -687,15 +688,12 @@ void ParsedFile::classlist(core::Context ctx, vector<string> &out) {
     }
 }
 
-void ParsedFile::subclasses(core::Context ctx, map<string, set<string>> &out) {
-    // Ignore files in dirs we don't care about
-    if (path.find("test/") == 0 || path.find("scripts/") == 0 || path.find("lib/identification/scripts/") == 0 ||
-        path.find("ruby_benchmarks/") == 0 || path.find("/test/") != string_view::npos ||
-        path.find("/benchmark/") != string_view::npos) {
-        // Ignoring 'test/' also ignores the Sorbet CLI test input files :(
-        if (!(path.find("test/cli/autogen-subclasses") == 0)) {
-            return;
-        }
+void ParsedFile::subclasses(core::Context ctx, vector<string> &absolutePathsToIgnore,
+                            vector<string> &relativePathsToIgnore,
+                            map<string, set<pair<string, Definition::Type>>> &out) {
+    // We prepend "/" to `path` to mimic how `isFileIgnored` gets called elsewhere
+    if (sorbet::FileOps::isFileIgnored("", fmt::format("/{}", path), absolutePathsToIgnore, relativePathsToIgnore)) {
```

Comment:
This is kinda hacky, but it seemed worth it to have my new `--autogen-subclasses-ignore` flag behave like the existing `--ignore` flag

Reply:
I didn't get what this does,

Reply:
[bigger comment](https://github.com/sorbet/sorbet/pull/1040/files#diff-fe69bef0ea7ee4e25c450060a16165f5R7)